### PR TITLE
Dev sir 917 vault policy engine

### PR DESCRIPTION
**Pull request by aushakov**

### CODE REVIEW

File: src/VaultApi.ApiContract/transaction-approval-requests.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+message CreateTransactionApprovalRequestResponse {
+  oneof body {
+    CreateTransactionApprovalRequestResponseBody response = 1;
+    CreateTransactionApprovalRequestErrorResponseBody error = 2;
+  }
+}
+
+message KeyKeeperRequest {
+  string key_id = 1;
```
Comment (KonstantinRyazantsev) [inline src/VaultApi.ApiContract/transaction-approval-requests.proto:46]: is it key keeper id?

File: src/VaultApi.Common/Persistence/TransactionApprovalConfirmations/TransactionApprovalConfirmationsRepository.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                .ToListAsync();
+        }
+
+        public async Task InsertOrIgnoreAsync(TransactionApprovalConfirmation transactionApprovalConfirmation)
+        {
+            await using var context = new DatabaseContext(_dbContextOptionsBuilder.Options);
+
+            try
+            {
+                context.TransactionApprovalConfirmations.AddRange(transactionApprovalConfirmation);
```
Comment (KonstantinRyazantsev) [inline src/VaultApi.Common/Persistence/TransactionApprovalConfirmations/TransactionApprovalConfirmationsRepository.cs:37]: why AddRange if there is only one entity?

File: src/VaultApi/GrpcServices/TransactionApprovalRequestsService.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                return GetErrorResponse(
+                    CreateTransactionApprovalRequestErrorResponseBody.Types.ErrorCode.InvalidParameters,
+                    "API key id required");
+            }
+
+            var tenantId = context.GetTenantId();
+
+            if (string.IsNullOrEmpty(tenantId))
+            {
+                tenantId = "unknown"; // TODO: add tenant id to the private vault api key
```
Comment (KonstantinRyazantsev) [inline src/VaultApi/GrpcServices/TransactionApprovalRequestsService.cs:66]: what is it needed for?

File: src/VaultApi.ApiContract/transaction-approval-requests.proto (hunk 1 above)
Comment (aushakov) [reply src/VaultApi.ApiContract/transaction-approval-requests.proto:46]: It's a key keeper public key hash. As we discussed before the better name is KeyId. This property uses to find key keeper on Sirius side.

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**